README.md

```diff
@@ -2,40 +2,206 @@
+        numsDoubled.forEach(System.out::println);
+    }
+}
+```
+
+The program prints the result of mapping:
+
+```text
+12
+-4
+14
+40
+```
+
+Let's map a list of strings to uppercase. We will
+pass a method reference as the mapper function:
 
 ```java
 import java.util.List;
 import java.util.Arrays;
 
 public class Example {
     public static void main(String[] args) {
-        List<String> names = Arrays.asList("John", "Jake", "Jill", "Kate", "Ezra", "Maddie");
-        List<String> namesUpper = names.stream()
-                                    .map(String::toUpperCase)
-                                    .toList();
-        System.out.println(namesUpper);
+        List<String> names = Arrays.asList("Amir", "Neela", "Dean");
+        names.stream()
+                .map(String::toUpperCase)
+                .forEach(System.out::println);
+    }
+}
+```
+
+The program prints the names in uppercase:
+
+```
+AMIR
+NEELA
+DEAN
+```
+
+The `Stream` interface has special mapping methods
+when working with a collection that wraps a primitive data type,
+such as `List<Integer>`. 
+
+- `DoubleStream	mapToDouble(ToDoubleFunction<? super T> mapper)`
+- `IntStream	mapToInt(ToIntFunction<? super T> mapper)`
+- `LongStream	mapToLong(ToLongFunction<? super T> mapper)`
+
+For example, if a stream is created from a list of integers, we use `mapToInt`:
+
+```java
+import java.util.List;
+import java.util.stream.IntStream;
+
+public class Example {
+    public static void main(String[] args) {
+        List<Integer> nums = List.of(6, -2, 7, 20);
+
+        //double each element in the stream
+        IntStream numsDoubled = nums.stream().mapToInt(i -> i*2);
+        numsDoubled.forEach(System.out::println);
+    }
+}
+```
+
+### Mapping a user defined class
+
+Let's explore how to map the `Employee` class:
+
+```java
+public class Employee {
+    private String name;
+    private double salary;
+    private int age;
+
+    public Employee(String name, double salary, int age) {
+        this.name = name;
+        this.salary = salary;
+        this.age = age;
+    }
+
+    public String getName() { return name; }
+
+    public double getSalary() { return salary; }
+
+    public int getAge() { return age; }
+    
+}
+```
+
+We can extract the employee names by passing the method reference
+`Employee::getName` as the mapper function:
+
+```java
+import java.util.List;
+
+public class Example {
+
+    public static void main(String[] args) {
+        List<Employee> employees = List.of(
+                new Employee("fred", 42000.0, 24),
+                new Employee("devon", 48000.0, 27),
+                new Employee("pat", 77000.0, 32));
+
+        //map names
+        employees.stream()
+                .map(Employee::getName)
+                .forEach(System.out::println);
     }
 }
 ```
 
+The program prints:
+
+```text
+fred
+devon
+pat
+```
+
+If we want to get the names in uppercase, we can either use a lambda
```

Comment:
Opinion: Possibly remove the word "either" or put the two examples (lambda vs chain calls) sequentially after the other with "Or we can chain a sequence of calls..." and then say "Both these will produce the following text" with the output.

Reply:
thanks, updated wording